### ISSUE TITLE
RIP Google Workspace

### ISSUE THREAD

**Issue by mr-handerson** [TechPerplexed/Gooby#220]
Just got the email... It was a good run while it lasted.

Wondering what others have decided to do?

Quick research so far seems to be group buys for Dropbox or Box. Not sure if I'm willing to go through all that effort at this point to potentially have it happen again in the future.

Might be time to build a new NAS and bring it back local.

**Comment by TechPerplexed** [comment]
Yeah, sadly it's happening. I haven't had the email yet, but I'm preparing. In fact we purchased Synology NAS (1821+) on Prime Day. It will probably be about 5 years to break even, if we decide to fill it to its max with eight 16TB drives. Our collection is not HUGE huge, so that would easily store everything.

As I mentioned to a friend today, the thing is that Google, although great in some ways, has also been a pain in the neck almost as often. During the days of setting up Plex, Emby and Jellyfin, we went through several weeks, sometimes months of almost non-stop API bans. Then there were the price hikes, we started with paying $10 and now it's $30, So I just can't see myself signing up for yet another cloud (be it Dropbox or something else) and go through the same nerve wracking experience again.

For now the NAS is perfect and since there are only the two of us, usually watching stuff together, the NAS even runs Plex without any issues.

It also means that right now I'm not using Omni because our ISP doesn't play nice with port 80. I know others, including Kelinger, uses it on their home setup without any issues, so I don't think it'll go anywhere ^_^

**Comment by mr-handerson** [comment]
I will admit that I was out of the loop and unaware that the purge had started. My little ecosystem has been humming along wonderfully, so I've just been using and enjoying. Seeing that email came as quite the shock! And I'm a little salty at the timing too. Just after Prime Day and too far away from Black Friday. Them's the breaks I guess...

That is a very nice purchase! You're not concerned about the lack of hardware transcoding because of the Ryzen? Or you think it will handle what you need without it?

I am of the same sentiment now that I've had a bit more time to mull it over. Not going through all the hassle of migrating to another cloud solution to potentially just go through all this again after an indeterminate amount of time.

So now I'm disappearing deep down the NAS rabbit hole. Given the costs for the level of off-the-shelf NAS I'd be looking at for my needs, it seems like I'll be building my own. That is all a little overwhelming at the moment, but I'm sure I'll make enough sense of it to be able to proceed in a timely fashion. First order of business is getting something built with enough storage so that I can at least offload everything from Google and bid them farewell. 

Appreciate the reply, and good that you've got things sorted in advance. Now you don't even need to wait for Google if you don't want to. Enjoy your nice new Syno!

**Comment by bifa2** [comment]
Sad times! In order to move to some local storage without impact to Plex, I think I'll have to mount the storage under `/mnt/google/` and then replicate my Google Drive folder structure (starting from the root, not "My Drive"), right?

**Comment by TechPerplexed** [comment]
Well, this past week has been quite the rabbit hole and learning curve, but I think I'm reaching the bottom at last (in a good way).

I have installed all my essential apps in Synology's own Docker, except I haven't managed to get Watchstate to work (yet). I set Plex to use more resources than the rest, and it's working great. We haven't encountered anything yet that doesn't play smoothly, but then again we never add anything beyond 1080p so no idea how 4K would work. I did however set almost every service with a scheduler not to run during our prime watching time, because especially the syncing from Google keeps the system busy enough to interfere with Plex.

I'm using the NAS' native app "Cloud Sync" to move everything across. When I say everything, I actually mean the essential movies and shows. Google's "unlimited" storage has allowed us to add stuff "just in case one of us would ever watch that", but of course now our strategy is changing to "do we really like and need this?" To that end I have created an additional root folder in each *arr and sync only those with the NAS. I'm transferring anything we feel we MUST keep (or are watching now) to the sync folder which then miraculously appears in the NAS after a while.

So far it's been working far better than expected, and it's nice to have an exit strategy once that Google axe drops on us as well.

**Comment by bifa2** [comment]
For minimal disruption/downtime, I'm wondering whether I can just repoint rclone to a SMB mount rather than google drive so that essentially Plex doesn't see any changes (once I've cloned the contents). Thoughts? 

Any idea how I would do this within Gooby itself? In otherwords, what I'd have to modify to try and stop it connecting to Google Drive? Would it work simply by editing the remote config file?

**Comment by bifa2** [comment]
Ok, first hurdle getting an rclone config to work from Linux to a Windows SMB share is done with:

```
[GDrive]
type = smb
host = hostname.mydomain.com
user = LocalUser
pass = removed
domain = COMPUTERNAME
```

Listing share contents works via (example) `rclone ls GDrive:SHARENAME /mnt`

but is there anything in the Gooby codebase that'd stop this from working? Perhaps the way it's configured to mount persistently with certain options? I'm not sure how that's done at the moment...

**Comment by TechPerplexed** [comment]
Sorry for the late reply - did you get it working? Afaik there isn't anything to prevent what you want. Meanwhile, our NAS is getting more and more appreciated... 😺

**Comment by mr-handerson** [comment]
Yes. I am guilty of the same. I've been rather indiscriminate in what I've both downloaded and kept. So in parallel to trying to determine everything I need to build my new rig (it's daunting!), I'm also cleaning house and going through and deleting anything that falls into the won't ever watch or rewatch categories. The added benefit there being that I'll have less to download from Google once I finally reach that stage.

@bifa2 you're well ahead of where I'm currently at, since I'm ditching my entire setup for something new. What you've mentioned seems like it should work though.

**Comment by bifa2** [comment]
@TechPerplexed Still copying my content off GDrive 😅 before I switching over fully, but mounting with `--daemon` also works.

Out of curiosity, if I want to do some testing by adding a new mount into the Plex container, which yaml file should be edited to make that stick around?
```
/var/local/Gooby/Docker/components/21-plex.yaml
/var/local/Gooby/Docker/docker-compose.yaml
```

Both? I'm guessing when the gooby menu is used to 'install' a component, it takes the individual file and appends it to the `docker-compose.yaml` file perhaps?

When the content is finally copied over I'll just revert to the original mount details but modify the rclone config file directly to use SMB not GDrive and hopefully the libraries won't notice anything.

**Comment by TechPerplexed** [comment]
Right, Gooby (and Omni) utilize a quite unique merging method to consolidate all the components into one yaml file. So the one that is actually used is the docker-compose.yml file.

**Comment by bifa2** [comment]
Managed a successful test last night, modding the docker-compose file to contain a new mount path, adding that into the .env file, running `docker compose up -d` to pick up said changes, adding the mount details to rclone config (using the wizard) and then rclone mounting the SMB share. I did have to restart the Plex container though to get it to see the files. Prior that that whilst it could see the mounted directory it just appeared empty. So I guess that explains why mounts need to be up, before the docker container starts! 

I don't want to hijack this issue with my technical questions so if I have further about moving from GDrive to SMB and tweaking the Gooby internals to cater to that, I'll start a new thread!

I do have one last question here though: I'd like to make a small donation because your support has always been A+ and really quick over the years :) Is it definitely going to you, and not the owner of Omni?

**Comment by mr-handerson** [comment]
Will close this up at this point.

I'm in the process of building an Unraid NAS and will move everything local. It's going to be quite the initial upfront expenditure, but then I hope to be set for the long run and won't have to concern myself with any of this again.

I'm also concurrently going through and deleting things I'd previously downloaded and kept, just because I could. 10TB+ and counting so far!

Hope whatever solution folks end up going with keeps them happy and streaming!
